### PR TITLE
Save images in the filesystem for host envelopes.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/envelope/boot-container-test.toit [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import expect show *
+import host.directory
+import host.file
+import host.pipe
+
+import .exit-codes
+import .util show EnvelopeTest with-test
+import .boot-container-source as container
+
```
Comment (kasperl) [inline tests/envelope/boot-container-test.toit:13]: Drop one newline.

File: system/extensions/host/run-image.toit [hunk 2]
```diff
@@ -306,15 +379,16 @@ handle-arguments arguments/List container-manager/ContainerManager -> none:
...
+    [startup-dir, bundled-dir, installed-dir].do: | dir/string |
+      is-startup-dir := dir == startup-dir
+      // Directories are not required to be there.
+      if not file.is-directory dir: continue.do
+      stream := directory.DirectoryStream dir
+      try:
+        while file-name/string? := stream.next:
+          path := "$dir/$file-name"
+          if file.is-file path:
+            add-image path existing-uuids --run-boot=is-startup-dir --run-critical=is-startup-dir
```
Comment (kasperl) [inline system/extensions/host/run-image.toit:392]:
```suggestion
          if not file.is-file path: continue
          add-image path existing-uuids --run-boot=is-startup-dir --run-critical=is-startup-dir
```

File: system/extensions/host/run-image.toit [hunk 3]
```diff
@@ -223,6 +227,75 @@ class Firmware:
...
+  Runs the given block but ensures that no committed image is saved to the
+    file system.
+  */
+  without-fs-backup-do [block]:
+    save-to-fs_ = false
+    try:
+      block.call
+    finally:
+      save-to-fs_ = true
+
```
Comment (kasperl) [inline system/extensions/host/run-image.toit:297]: Drop one newline.

**Review by kasperl — APPROVED** [review]
Looks good to me!

File: system/extensions/host/run-image.toit (hunk 3 above)
Comment (floitsch) [reply system/extensions/host/run-image.toit:297]: done.

File: system/extensions/host/run-image.toit (hunk 2 above)
Comment (floitsch) [reply system/extensions/host/run-image.toit]: done.

File: tests/envelope/boot-container-test.toit (hunk 1 above)
Comment (floitsch) [reply tests/envelope/boot-container-test.toit:13]: done.

**Pull request merged**